pyproject.toml

```diff
@@ -24,44 +28,58 @@ classifiers = [
     "License :: OSI Approved :: BSD License",
     "Operating System :: OS Independent",
     "Programming Language :: Python",
-    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
     "Topic :: Scientific/Engineering :: Artificial Intelligence"
 ]
-requires-python = ">=3.6"
+requires-python = ">=3.9,<3.14"
 dependencies = [
-    "numpy",
-    "scikit-base",
-    "scipy",
+    "numpy<3",
+    "scikit-base<0.13.0",
+    "scipy<2.0.0",
 ]
 
 [project.optional-dependencies]
 
 tests = [
-    "pytest"
+    "pytest<8.4"
```

Comment:
I think we can leave `pytest` without bounds.

This is only affecting CI, not user experience, and `pytest` is usually very robust.